### PR TITLE
changed np.rollaxis with np.moveaxis in coordinates

### PR DESCRIPTION
**Pull request by Juan-789**
### Description

This pull request is to address ...

Address `coordinates` portion for #15721

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

**Comment by neutrinoceros** [comment]
Hi @Juan-789, thank you for you work !
Note that changing the name of the function to be called isn't sufficient as they have different signatures, and in fact we're seeing that the change as is breaks some tests.
See https://github.com/astropy/astropy/pull/15727#discussion_r1425151963 for guidance.

File: astropy/coordinates/tests/test_representation_methods.py [hunk 1]
```diff
@@ -367,14 +367,14 @@ def test_atleast_3d(self):
 
     def test_move_axis(self):
         # Goes via transpose so works without __array_function__ as well.
-        s0_10 = np.moveaxis(self.s0, 0, 1)
+        s0_10 = np.moveaxis(self.s0.data, 1, 0)
```
Comment (dhomeier) [inline astropy/coordinates/tests/test_representation_methods.py:370]: Why was this changed?

File: astropy/coordinates/tests/test_representation_methods.py [hunk 2]
```diff
@@ -367,14 +367,14 @@ def test_atleast_3d(self):
...
         assert s0_10.shape == (self.s0.shape[1], self.s0.shape[0])
         assert np.all(representation_equal(self.s0.T, s0_10))
         assert np.may_share_memory(s0_10.lon, self.s0.lon)
 
     def test_roll_axis(self):
         # Goes via transpose so works without __array_function__ as well.
-        s0_10 = np.rollaxis(self.s0, 1)
+        s0_10 = np.moveaxis(self.s0, 1)
```
Comment (dhomeier) [inline astropy/coordinates/tests/test_representation_methods.py:377]:
```suggestion
        s0_10 = np.moveaxis(self.s0, 1, 0)
```
`destination` is a required argument to `moveaxis`. I _think_ this should be the correct setting for the default `start=0`.

**Review by dhomeier — CHANGES_REQUESTED** [review]
I suggest to look into the `start` documentation section of https://numpy.org/doc/stable/reference/generated/numpy.rollaxis.html to ensure the replacement parameters are indeed correctly set.

**Comment by eerovaher** [comment]
The list of sub-packages in #15721 was created by simply looking which sub-packages contain the string `rollaxis(` anywhere in their source code, but seeing the one occurrence of `np.rollaxis()` in coordinates in its context makes it clear that it should not be replaced. Although NumPy recommends using `np.moveaxis()` instead of `np.rollaxis()`, the latter is not deprecated and it is good to test that it still works, and that's exactly what `test_roll_axis()` does. Properly replacing `np.rollaxis()` with `np.moveaxis()` would make `test_roll_axis()` a duplicate of `test_move_axis()`, so even if we didn't want this test anymore the correct thing to do would be to remove the test entirely instead of updating it.

If you are interested in contributing to `astropy` then I suggest you have a look at #14818.
